src/node_crypto.cc

```diff
@@ -741,6 +743,56 @@ void SecureContext::AddCRL(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void SecureContext::UseExtraCaCerts(const char* fname) {
+  extra_root_certs_file = fname;
+}
+
+static void emitWarning(Environment* env, const char* fmt, ...) {
+  char warning[1024];
+  va_list ap;
+
+  va_start(ap, fmt);
+  vsnprintf(warning, sizeof(warning), fmt, ap);
+  va_end(ap);
+
+  Local<Object> process = env->process_object();
+  Local<v8::Value> f = process->Get(
+    FIXED_ONE_BYTE_STRING(env->isolate(), "emitWarning"));
+  Local<Value> args[] = {
+    node::OneByteString(env->isolate(), warning, strlen(warning))
+  };
+
+  f.As<v8::Function>()->Call(process, arraysize(args), args);
```

Comment:
@mhdawson 's link above says

> The latter (MakeCallback) case calls the JS callback function, but then also ends the current tick, calling process._tickCallback() and any other callbacks registered via nextTick(). This form is intended only for use by C++ code that was dispatched from the libuv event loop, and not for C++ code that was invoked from JS code.

It sounds like `MakeCallback()` has lots of side effects, are you sure its right? My usage seems to fit the "not C++ invoked from the event loop" case.

It occured to me that `process.emitWarning()` could actually emit... which would call user code, but I checked, and it emits in the next tick, so isn't going to hit any user-supplied js.

I don't know enough about the distinction to choose, I'll need some guidance here.